pkg/api/types.go

```diff
@@ -742,6 +746,31 @@ type VsphereVirtualDiskVolumeSource struct {
 	FSType string `json:"fsType,omitempty"`
 }
 
+type AzureDataDiskCachingMode string
+
+const (
+	AzureDataDiskCachingNone      AzureDataDiskCachingMode = "None"
+	AzureDataDiskCachingReadOnly  AzureDataDiskCachingMode = "ReadOnly"
+	AzureDataDiskCachingReadWrite AzureDataDiskCachingMode = "ReadWrite"
+)
+
+// AzureDisk represents an Azure Data Disk mount on the host and bind mount to the pod.
+type AzureDiskVolumeSource struct {
+	// Data Disk Name
+	DiskName string `json:"diskName"`
+	// Data Disk URI
+	DataDiskURI string `json:"diskURI"`
+	// Host Caching mode: None, Read Only, Read Write.
+	CachingMode AzureDataDiskCachingMode `json:"cachingMode,omitempty"`
```

Comment:
Spec what happens if this is not specified.

Since this is omitempty, it is optional, so it should be a pointer.